### DISCUSSION TITLE
Broadcast kitten: map to match current tab

### DISCUSSION THREAD

**Discussion by aorith** [kovidgoyal/kitty#3183] [Q&A] [answered]
Hi all,

I'm trying to create a mapping to trigger the new Broadcast kitten matching only the current tab windows.
AFAIK there is an env variable pointing to the current kitty window id but not the current tab id.

At this moment I'm using a "kitten wrapper" but I'm sure there is a better solution:

`map kitty_mod+s kitten kittens/broadcast_to_current_tab.py`

```python
# broadcast_to_current_tab.py
from kittens.tui.handler import result_handler


def main(args):
    pass


@result_handler(no_ui=True)
def handle_result(args, answer, target_window_id, boss):
    tab = boss.active_tab
    boss.launch(
        "--allow-remote-control",
        "--type=os-window",
        "--title=Broadcast",
        "/bin/bash",
        "-lc",
        f"kitty +kitten broadcast --match-tab=id:{tab.id}",
    )
```

Many thanks in advance!

**Comment by kovidgoyal** [comment, marked as answer]
Just use --match-tab and specify the window id, this will select the tab containing the window. The only issue is if there is also a tab with the same id, in that case it will select that tab instead, but this should be relatively rare.  If that is a problem for you, it should be an easy patch to add window_id based matching to --match-tab.

**Comment by aorith** [comment]
Hey!

I just saw your last commit, that simplifies it a lot. Thanks!

**Comment by Felixoid** [comment]
Worth mentioning the recent improvement. `--match-tab` accepts now `recent:0` argument for the current tab.
